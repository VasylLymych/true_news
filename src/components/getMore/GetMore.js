import React from "react";
import s from "./GetMore.module.css"

const GetMore = props => {

    return <div onClick={() => {
        props.getMoreNews(props.pageSize)

    }} className={s.getMoreWrapper}>Загрузити ще</div>
}

export default GetMore