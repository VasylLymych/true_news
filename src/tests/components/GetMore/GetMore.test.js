import GetMore from "../../../components/getMore/GetMore"
import {create} from "react-test-renderer"

let component;
const func = jest.fn()
const setUp = () => create(<GetMore getMoreNews={func}/>)

describe("GetMore component test", () => {
    beforeEach(() => {
        return component = setUp()
    })
    test("component snapshot", () => {
        expect(component).toMatchSnapshot()
    })
    test("onClick event", () => {
        component.root.findByType("div").props.onClick();
        expect(func.mock.calls.length).toBe(1)
    })
})