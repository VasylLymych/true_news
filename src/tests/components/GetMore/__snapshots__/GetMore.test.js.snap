// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetMore component test component snapshot 1`] = `
<div
  className="getMoreWrapper"
  onClick={[Function]}
>
  Загрузити ще
</div>
`;

exports[`GetMore component test onClick event 1`] = `
<div
  className="getMoreWrapper"
  onClick={[Function]}
>
  Загрузити ще
</div>
`;
